### PR TITLE
Category filter-sort-search

### PR DESCRIPTION
**Pull request by jul4ajka**

### CODE REVIEW

**Review by moran711 — DISMISSED** [review]

**Review by AndriyLymych — APPROVED** [review]

File: helpers/helper.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+class FilterHelper {
+  filterItems(args = {}) {
+    const filter = {};
+    const { roles, days, banned, category, search } = args;
+
+    if (roles && roles.length) {
+      filter.role = { $in: roles };
+    }
+
+    if (category && category.length) {
```
Comment (moran711) [inline helpers/helper.js:11]: I think that you nedd to pass here `_id` param

File: modules/category/category.graphql.js [hunk 2]
```diff
@@ -14,13 +15,41 @@ const categoryType = `
...
     code: String
     name: [LanguageInput]
     images: ImageSetInput
     available: Boolean
-  }`;
+	}`;
+
+const FilterInputComponent = `
+  input FilterInputComponent {
+    roles: [String!]
```
Comment (moran711) [inline modules/category/category.graphql.js:35]: Please preatify this.

File: modules/category/category.graphql.js [hunk 3]
```diff
@@ -14,13 +15,41 @@ const categoryType = `
...
+		banned: [Boolean]
+		category: [ID]
+    search: String
+}
+`;
+
+const SortInputComponent = `
+	input SortInputComponent {
+		name: Int
```
Comment (moran711) [inline modules/category/category.graphql.js:44]: same

**Review by AndriyLymych — APPROVED** [review]

File: helpers/helper.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+const mongoose = require('mongoose');
+class FilterHelper {
```
Comment (AndriyLymych) [inline helpers/helper.js:2]: move this line to line №3

File: typeDefs.js [hunk 5]
```diff
@@ -494,8 +502,10 @@ const typeDefs = gql`
   ${LoginInput}
   ${userRegisterInput}
   ${businessTextInput}
-  ${userFilterInput}
-  ${userSortInput}
+	${userFilterInput}
+	${userSortInput}
+	${FilterInputComponent}
+	${SortInputComponent}
   ${adminConfirmInput}
```
Comment (AndriyLymych) [inline typeDefs.js:509]: please pretify this

**Review by AndriyLymych — APPROVED** [review]

**Review by kapoliub — APPROVED** [review]

**Review by moran711 — APPROVED** [review]

**Pull request merged**